test/e2e/form-validation.test.js

```diff
@@ -4,48 +4,62 @@ const VlFormValidationPage = require('./pages/vl-form-validation.page');
     beforeEach(async () => {
         return vlFormValidationPage.load();
     });
 
-    async function assertThatGeenFoutmeldingWordtGetoond(formValidationElement, validationMessageElement) {
-        await assert.eventually.equal(formValidationElement.getErrorClass(), expectedErrorClass);
-        await assert.eventually.equal(formValidationElement.getSuccessClass(), expectedSuccessClass);
-        await assert.eventually.isFalse(formValidationElement.hasClass(expectedErrorClass));
-        await assert.eventually.isFalse(formValidationElement.hasClass(expectedSuccessClass));
+    async function assertThatGeenFoutmeldingWordtGetoondBijGeldigeInput(formValidationElement, validationMessageElement, setGeldigeInputFunctie) {
+        if (setGeldigeInputFunctie) {
+            await setGeldigeInputFunctie();
+        }
+        await assert.eventually.isFalse(formValidationElement.hasError());
+        await assert.eventually.isFalse(formValidationElement.isSuccess());
         await assert.eventually.isFalse(validationMessageElement.isDisplayed());
     }
 
-    async function assertThatFoutmeldingenCorrectGetoondWorden(formValidationElement, validationMessageElement) {
-        await assert.eventually.isTrue(formValidationElement.hasClass(expectedErrorClass));
-        await assert.eventually.isFalse(formValidationElement.hasClass(expectedSuccessClass));
+    async function assertThatFoutmeldingenCorrectGetoondWordenBijOngeldigeInput(formValidationElement, validationMessageElement, setOngeldigeInputFunctie) {
+        if (setOngeldigeInputFunctie) {
+            await setOngeldigeInputFunctie();
+        }
+        await assert.eventually.isTrue(formValidationElement.hasError());
+        await assert.eventually.isFalse(formValidationElement.isSuccess());
         await assert.eventually.isTrue(validationMessageElement.isDisplayed());
         await assert.eventually.equal(formValidationElement.getErrorPlaceholder(), await validationMessageElement.getErrorId());
         await assert.eventually.equal(validationMessageElement.getText(), await formValidationElement.getErrorMessage());
     }
 
     async function assertThatFormMetInputFieldCorrectValideert(form, geldigeInput, ongeldigeInput) {
-        await assert.eventually.isTrue(form.hasAttribute('data-vl-validate-form'));
-
         const inputField = await form.getInputField();
         await assert.eventually.isTrue(inputField.isRequired());
 
-        const validationMessageElement = await form.getErrorMessage();
-
-        await assertThatGeenFoutmeldingWordtGetoond(inputField, validationMessageElement);
+        await assertThatFormValidationCorrectValideert(inputField, await form.getErrorMessage(),
+            async function() {
+                await inputField.setInputValue(geldigeInput);
+                await inputField.blur();
+            },
+            async function() {
+                await inputField.setInputValue(ongeldigeInput);
+                await inputField.blur();
```

Comment:
nog duplicatie in deze 2. misschien een functie die een functie teruggeeft maken